src/main/java/org/kohsuke/github/GHRepository.java

```diff
@@ -3257,29 +3256,13 @@ void populate() throws IOException {
             return; // can't populate if the root is offline
         }
 
-        final URL url = requireNonNull(getUrl(), "Missing instance URL!");
+        // We don't take the URL provided in the JSON as it is not reliable in some cases:
+        // There is bug in Push event payloads that returns the wrong url.
+        // All other occurrences of "url" take the form "https://api.github.com/...".
+        // For Push event repository records, they take the form "https://github.com/{fullName}".
+        // And also for Installation event payloads, the URL is not provided at all.
 
-        try {
-            // IMPORTANT: the url for repository records does not reliably point to the API url.
-            // There is bug in Push event payloads that returns the wrong url.
-            // All other occurrences of "url" take the form "https://api.github.com/...".
-            // For Push event repository records, they take the form "https://github.com/{fullName}".
-            root().createRequest()
-                    .withPreview(BAPTISTE)
-                    .withPreview(NEBULA)
-                    .setRawUrlPath(url.toString())
-                    .fetchInto(this);
-        } catch (HttpException e) {
-            if (e.getCause() instanceof JsonParseException) {
-                root().createRequest()
-                        .withPreview(BAPTISTE)
-                        .withPreview(NEBULA)
-                        .withUrlPath("/repos/" + full_name)
-                        .fetchInto(this);
-            } else {
-                throw e;
-            }
-        }
+        root().createRequest().withPreview(BAPTISTE).withPreview(NEBULA).withUrlPath(getApiTailUrl("")).fetchInto(this);
```

Comment:
Decided to simplify that but I can add a bunch of ifs if you prefer using the URL coming from the JSON when it is around.

Reply:
No, this is functional and clean. 💯